python/ray/serve/_private/deployment_state.py

```diff
@@ -1368,34 +1407,35 @@ def _set_target_state(self, target_info: DeploymentInfo) -> None:
             ):
                 ServeUsageTag.NUM_REPLICAS_LIGHTWEIGHT_UPDATED.record("True")
 
-        self._target_state = target_state
-        self._curr_status_info = DeploymentStatusInfo(
-            self.deployment_name, DeploymentStatus.UPDATING
-        )
-        self._replica_constructor_retry_counter = 0
-        self._backoff_time_s = 1
-
-        app_msg = f" in application '{self.app_name}'" if self.app_name else ""
-        logger.info(
-            f"Deploying new version of deployment {self.deployment_name}{app_msg}."
-        )
-
-    def _set_target_state_autoscaling(self, num_replicas: int) -> None:
-        """Update the target number of replicas based on an autoscaling decision.
-
-        This differs from _set_target_state because we are updating the
-        target number of replicas base on an autoscaling decision and
-        not a redeployment. This only changes the target num_replicas,
-        and doesn't change the current deployment status.
-        """
-
-        new_info = copy(self._target_state.info)
-        new_info.set_autoscaled_num_replicas(num_replicas)
-        new_info.version = self._target_state.version.code_version
-
-        target_state = DeploymentTargetState.from_deployment_info(new_info)
+        # Determine whether the updated target state is simply scaling
+        is_scale = target_state.compare_state_excluding_num_replicas(self._target_state)
+        # If deployment is not in the middle of updating, and this new
+        # target state only changes the number of replicas, then set
+        # status to be UPSCALING/DOWNSCALING
+        if self._target_state.info and is_scale:
+            if self._curr_status_info.status != DeploymentStatus.UPDATING:
+                new, old = (target_state.num_replicas, self._target_state.num_replicas)
+                # New replicas should never be equal to old replicas because both
```

Comment:
Is this still true, now that the number of replicas can be scaled by `target_capacity`?

E.g. if you apply a new config with the same number of replicas, but a smaller `target_capacity`, then does the deployment transition to `DOWNSCALING`?

Reply:
The comment is still true even with `target_capacity` because `target_capacity` does not affect the target state, it is used in the read-only path (e.g. the scaling method reads `target_state.num_replicas * target_capacity`).

If you apply a new config with num replicas the same + a lower `target_capacity`, the status doesn't change. This is a good thing to bring up, I think we should have an offline discussion about what the status should transition to in this case (will not be addressing in this PR).